src/main/java/com/zendesk/maxwell/MaxwellConfig.java

```diff
@@ -124,52 +123,74 @@ private String parseLogLevel(String level) {
 		return level;
 	}
 
+
+	private String fetchOption(String name, OptionSet options, Properties properties, String defaultVal) {
+		if ( options.has(name) )
+			return (String) options.valueOf(name);
+		else if ( (properties != null) && properties.containsKey(name) )
+			return (String) properties.getProperty(name);
+		else
+			return defaultVal;
+	}
+
+	private MaxwellMysqlConfig parseMysqlConfig(String prefix, OptionSet options, Properties properties) {
+		MaxwellMysqlConfig config = new MaxwellMysqlConfig();
+		config.host     = fetchOption(prefix + "host", options, properties, null);
+		config.password = fetchOption(prefix + "password", options, properties, null);
+		config.user     = fetchOption(prefix + "user", options, properties, null);
+		config.port     = Integer.valueOf(fetchOption(prefix + "port", options, properties, "3306"));
+		config.setJDBCOptions(fetchOption(prefix + "jdbc_options", options, properties, null));
+		return config;
+	}
+
 	private void parse(String [] argv) {
 		OptionSet options = buildOptionParser().parse(argv);
 
+		Properties properties;
+
 		if ( options.has("config") ) {
-			parseFile((String) options.valueOf("config"), true);
+			properties = parseFile((String) options.valueOf("config"), true);
 		} else {
-			parseFile(DEFAULT_CONFIG_FILE, false);
+			properties = parseFile(DEFAULT_CONFIG_FILE, false);
 		}
 
 		if ( options.has("help") )
 			usage("Help for Maxwell:");
 
-		if ( options.has("log_level")) {
-			this.log_level = parseLogLevel((String) options.valueOf("log_level"));
-		}
-
-		this.maxwellMysql.parseOptions("", options);
+		this.log_level = fetchOption("log_level", options, properties, null);
```

Comment:
could go down that road, but the right there would be to be like:

```
  new MaxwellConfigOptionParser(this, options, properties).parse()
```

or something.  Leaving instance vars around just to not pass them into other methods (ie they'll get thrown away at the end of the method) is IMO kinda bad juju.